source/Fleet.cpp

```diff
@@ -87,14 +87,10 @@ void Fleet::Load(const DataNode &node)
 			government = GameData::Governments().Get(child.Token(1));
 		else if(key == "names" && hasValue)
 			names = GameData::Phrases().Get(child.Token(1));
-		else if(key == "fighters" && hasValue)
-		{
-			fighterNames = GameData::Phrases().Get(child.Token(1));
-			child.PrintTrace("Warning: Deprecated use of \"fighters\" <names>."
-				" Use \"names\" <names> node under \"fighters\" instead.");
-		}
-		else if(key == "fighters" && child.HasChildren())
+		else if(key == "fighters" && (hasValue || child.HasChildren()))
 		{
+			if(hasValue)
+				fighterNames = GameData::Phrases().Get(child.Token(1));
 			for(const DataNode &grand : child)
```

Comment:
it's not an issue if we dont have children and we do this, right?

Reply:
begin() == end() in the case of an empty list, which means the for each loop never runs. (For each loops are basically just shorthand for `for(auto it = list.begin(); it != list.end(); ++it)`.)